modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -289,7 +295,18 @@ def run_host(ip)
         smb_login
 
         begin
-          shares = simple.client.net_share_enum_all(ip)
+          # Return all shares if `Share` option has not been set
+          if datastore['Share'].nil?
+            shares = simple.client.net_share_enum_all(ip)
+          else
+            # Return specific share if the `Share` option has been set
+            simple.client.net_share_enum_all(ip).each { |share| shares = [share] if share[:name] == datastore['Share'] }
+            # Return an error and all shares if `Share` option has been set but no matches were found
+            if shares.empty?
+              print_error("No shares match #{datastore['Share']} - returning all shares instead")
+              shares = simple.client.net_share_enum_all(ip)
+            end
```

Comment:
Is there an existing API that we can call instead here? i.e. An existing function that takes a single share to connect to? Or is there a way to avoid this lookup entirely?

Reply:
I looked into using [`tree_connect`](https://github.com/rapid7/ruby_smb/blob/95ffce90f3fbd2b2d8d00b643e318fc38cce52bd/lib/ruby_smb/client.rb#L606) but it didn't seem like what I was after, but maybe that isn't what you're referring to. I will spend some more time looking into that again today.

Reply:
This isn't a blocker for me